go/vt/vtgate/tx_conn.go

```diff
@@ -81,6 +81,7 @@ func (txc *TxConn) Commit(ctx context.Context, session *SafeSession) error {
 func (txc *TxConn) commitNormal(ctx context.Context, session *SafeSession) error {
 	if err := txc.runSessions(session.PreSessions, func(s *vtgatepb.Session_ShardSession) error {
 		defer func() { s.TransactionId = 0 }()
+		// TODO(deepthi): this Commit needs to happen on the right tablet (that the transactionID belongs to)
```

Comment:
Is this correct? or should we be checking for tabletType here and doing the commit only if it is MASTER?

Reply:
we need to support commit/rollback for replica transactions and direct them to the correct replica. Fixed.